pre_commit_hooks/check_added_large_files.py

```diff
@@ -21,6 +22,18 @@ def lfs_files() -> Set[str]:
     return set(json.loads(lfs_ret)['files'])
 
 
+def lfs_all_files() -> Set[str]:
+    """All files in the repo tracked by git-lfs."""
+    try:
+        # This also includes all empty files in the repository:
+        # https://github.com/git-lfs/git-lfs/issues/4660
+        lfs_ret = cmd_output('git', 'lfs', 'ls-files', '--name-only')
```

Comment:
Sorry for the delay. I've been investigating alternatives here, but I couldn't come up with a good one.

The main issue that I'm concerned with is that it is relatively easy to end up with a file that is marked with `filter=lfs` in gitattributes but is not actually using git-lfs:
- If you don't have git-lfs installed, then git will just ignore the lfs filter and will commit as a regular file. 
- If you have it installed, but you edit gitattributes directly instead of using `git lfs track` and you don't `git add --renormalize`, then the files are also committed as regular files.

As an aside - it would be great to have a pre-commit check that catches these cases, checking that all files that have `filter=lfs` in gitattributes are actually being tracked by git-lfs. If you know a good way of enforcing that, then it would be easier to switch this to something else based on `git check-attr`

Reply:
hmmm, yeah that's tricky -- given `git-lfs` can't produce a parseable output I don't think we can use that.  I think the attributes would be a good compromise though, despite the ways to get around it.

Reply:
Sounds good; re-implemented using check-attr instead. PTAL

Do you think it is worth keeping the old `lfs_files` function, or should I switch both paths (with and without `--enforce-all`) to use the new approach?

Reply:
yeah let's switch both of them to be consistent

Reply:
Done